homeassistant/util/yaml/loader.py

```diff
@@ -116,7 +116,11 @@ def _include_yaml(loader: SafeLineLoader, node: yaml.nodes.Node) -> JSON_TYPE:
 
     """
     fname = os.path.join(os.path.dirname(loader.name), node.value)
-    return _add_reference(load_yaml(fname), loader, node)
+    try:
+        return _add_reference(load_yaml(fname), loader, node)
+    except FileNotFoundError as exc:
+        _LOGGER.error("Unable to read file %s: %s", fname, exc)
+        raise HomeAssistantError(exc)
```

Comment:
Better to just raise the exception with the formatted string you are logging.

Whatever catches that exception further out should log it.

Reply:
So, not passing on the original exception?

Reply:
Nope not here. HomeAssistantError(Exception) takes a string as first argument.

You can do:
```
raise HomeAssistantError("formatted {}".format("string") from exc
```

If you want the exception trail to remain. But not needed there imho.

Reply:
Also. You might want to add some data from the yaml node. Maybe it includes row number?

Reply:
Great idea!